src/dataflow/src/source/postgres.rs

```diff
@@ -130,12 +134,46 @@ impl PostgresSourceReader {
     ) -> Self {
         Self {
             source_name: source_name.clone(),
+            source_tables: HashMap::from_iter(
+                connector
+                    .details
+                    .tables
+                    .iter()
+                    .map(|t| (t.relation_id, t.clone())),
+            ),
             connector,
             lsn: 0.into(),
             metrics: PgSourceMetrics::new(metrics, source_name),
         }
     }
 
+    /// Validates that all expected tables exist in the publication tables and they have the same schema
+    fn validate_tables(&self, tables: Vec<TableInfo>) -> Result<(), anyhow::Error> {
+        let pub_tables: HashMap<u32, PostgresTable> = tables
+            .into_iter()
+            .map(|t| (t.rel_id, PostgresTable::from(t)))
+            .collect();
+        for (id, schema) in self.source_tables.iter() {
+            if let Some(pub_schema) = pub_tables.get(id) {
```

Comment:
This would be better written as a `match` statement since you handle both this and the `else` case